### PR TITLE
Add 245 and 512 Breakout Boards

### PR DESCRIPTION
**Pull request by Stemby**
##### SUMMARY
Added 245 and 512 Breakout Boards (analog joysticks).

##### OPEN QUESTIONS / UNRESOLVED ISSUES
Packages are sketchy.

##### CHECKLIST

- [X] I have read and followed the [library conventions](https://docs.librepcb.org/#libraryconventions).
- [ ] For packages, I followed IPC7351C (see details in library conventions).
- [X] I'm the copyright owner of the added content (i.e. the changes are made by myself, not copied/imported from somewhere else).
- [X] I agree to publish all my changes under the [CC0 Public Domain License](https://en.wikipedia.org/wiki/CC0), allowing everyone to use and modify the content without any restrictions.

### CODE REVIEW

File: pkg/df89cc5c-aef5-440c-bf14-13ed199bd265/package.lp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+(librepcb_package df89cc5c-aef5-440c-bf14-13ed199bd265
+ (name "Adafruit 512 Breakboard")
```
Comment (dbrgn) [inline pkg/df89cc5c-aef5-440c-bf14-13ed199bd265/package.lp:2]: According to the IPC standard, this should be:

```suggestion
 (name "ADAFRUIT_512")
```

File: dev/8954aa9e-7042-4ba8-aac4-cff090493d5b/device.lp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(librepcb_device 8954aa9e-7042-4ba8-aac4-cff090493d5b
+ (name "Adafruit 245 Breakout")
```
Comment (dbrgn) [inline dev/8954aa9e-7042-4ba8-aac4-cff090493d5b/device.lp:2]: According to the IPC standard, this should be

```suggestion
 (name "ADAFRUIT_245")
```

File: pkg/df89cc5c-aef5-440c-bf14-13ed199bd265/package.lp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+   (width 0.2) (fill false) (grab_area false)
+   (vertex (position -20.32 20.32) (angle 0.0))
+   (vertex (position 20.32 20.32) (angle 0.0))
+   (vertex (position 20.32 -20.32) (angle 0.0))
+   (vertex (position -20.32 -20.32) (angle 0.0))
+   (vertex (position -20.32 20.32) (angle 0.0))
+  )
+  (stroke_text d2db1600-1c07-4431-b2b8-819460eb997a (layer top_names)
+   (height 1.0) (stroke_width 0.2) (letter_spacing auto) (line_spacing auto)
+   (align center bottom) (position 0.0 22.86) (rotation 0.0)
```
Comment (dbrgn) [inline pkg/df89cc5c-aef5-440c-bf14-13ed199bd265/package.lp:43]: I'd move this down half a step:

```suggestion
   (align center bottom) (position 0.0 21.59) (rotation 0.0)
```

File: pkg/df89cc5c-aef5-440c-bf14-13ed199bd265/package.lp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+   (auto_rotate true) (mirror false) (value "{{NAME}}")
+  )
+  (stroke_text c70c9c82-93b3-44de-ae2c-f9b4931cc0e1 (layer top_values)
+   (height 1.0) (stroke_width 0.2) (letter_spacing auto) (line_spacing auto)
+   (align center top) (position 0.0 -22.86) (rotation 0.0)
```
Comment (dbrgn) [inline pkg/df89cc5c-aef5-440c-bf14-13ed199bd265/package.lp:48]: I'd move this up half a step:

```suggestion
   (align center top) (position 0.0 -21.59) (rotation 0.0)
```

**Review by dbrgn — CHANGES_REQUESTED** [review]
Thanks @Stemby for the contribution, and sorry for the loong review delay!

- Symbols:
  - They look good to me, but I wonder if they should be moved to the base library. What do you think, @ubruhin?
  - Additionally, I'd move VCC/GND to the left side (VCC on top, GND at the bottom) and the outputs to the right side.
- Components:
  - Same thing here, should they be moved to the base lib? @ubruhin 

Regarding the package, the footprint of the 245 looks totally different to me...

PCB:

img

Package:

img

**Comment by ubruhin** [comment]
> Symbols:
>        They look good to me, but I wonder if they should be moved to the base library. What do you think, @ubruhin?
>        Additionally, I'd move VCC/GND to the left side (VCC on top, GND at the bottom) and the outputs to the right side.
>    Components:
>        Same thing here, should they be moved to the base lib? @ubruhin

Yeah, since these look very generic, I think that makes sense! But I'd suggest some small changes:
- Maybe make the signal names slightly more expressive, at least the `S` could be renamed to `SEL` or `BTN` or so.
- Change the default value from `{{PARTNUMBER or DEVICE or COMPONENT}}` to `{{PARTNUMBER or DEVICE}}` since the component name is too generic to be useful in a schematic.

@Stemby Do you like to apply these changes and move the symbols and components to the base library? Or should we take care of that?

**Comment by rnestler** [comment]
@Stemby are you still active on this?

**Comment by ubruhin** [comment]
I extracted the sym/cmp to the base library: https://github.com/LibrePCB-Libraries/LibrePCB_Base.lplib/pull/131

Now closing this due to inactivity (the packages would require some rework to get them finished).

**Pull request closed without merging**